services/server/src/server/apiv1/verification/private/stateless/customReplaceMethods.ts

```diff
@@ -0,0 +1,164 @@
+import { VerificationExport } from "@ethereum-sourcify/lib-sourcify";
+import { PoolClient } from "pg";
+import {
+  bytesFromString,
+  getDatabaseColumnsFromVerification,
+} from "../../../../services/utils/database-util";
+
+export type CustomReplaceMethod = (
+  poolClient: PoolClient,
+  verification: VerificationExport,
+) => Promise<void>;
+
+export const replaceCreationInformation: CustomReplaceMethod = async (
+  poolClient: PoolClient,
+  verification: VerificationExport,
+) => {
+  if (
+    verification.status.creationMatch !== "perfect" &&
+    verification.status.creationMatch !== "partial"
+  ) {
+    throw new Error(
+      "Creation match is null, cannot replace creation information",
+    );
+  }
+
+  // Get database columns from verification
+  const databaseColumns =
+    await getDatabaseColumnsFromVerification(verification);
+
+  // Get existing verified contract to find deployment_id
+  const existingVerifiedContractQuery = `
+    SELECT vc.id, vc.deployment_id, cd.chain_id, cd.address, cd.contract_id
+    FROM verified_contracts vc
+    JOIN contract_deployments cd ON cd.id = vc.deployment_id
+    LEFT JOIN sourcify_matches sm ON sm.verified_contract_id = vc.id
+    WHERE cd.chain_id = $1 AND cd.address = $2
+    LIMIT 1
+  `;
+
+  const existingResult = await poolClient.query(existingVerifiedContractQuery, [
+    verification.chainId.toString(),
+    bytesFromString(verification.address),
+  ]);
+
+  if (existingResult.rows.length === 0) {
+    throw new Error(
+      `No existing verified contract found for address ${verification.address} on chain ${verification.chainId}`,
+    );
+  }
+
+  const existingVerifiedContract = existingResult.rows[0];
+  const deploymentId = existingVerifiedContract.deployment_id;
+
+  // Insert new creation code if it exists
+  let newCreationCodeHash: Buffer | undefined;
+  if (databaseColumns.onchainCreationCode) {
+    const creationCodeInsertResult = await poolClient.query(
+      `INSERT INTO code (code_hash, code, code_hash_keccak) 
+       VALUES (digest($1::bytea, 'sha256'), $1::bytea, $2) 
+       ON CONFLICT (code_hash) DO NOTHING 
+       RETURNING code_hash`,
+      [
+        databaseColumns.onchainCreationCode.bytecode,
+        databaseColumns.onchainCreationCode.bytecode_hash_keccak,
+      ],
+    );
+
+    if (creationCodeInsertResult.rows.length === 0) {
+      // Code already exists, get the hash
+      const existingCodeResult = await poolClient.query(
+        `SELECT code_hash FROM code WHERE code_hash = digest($1::bytea, 'sha256')`,
+        [databaseColumns.onchainCreationCode.bytecode],
+      );
+      newCreationCodeHash = existingCodeResult.rows[0].code_hash;
+    } else {
+      newCreationCodeHash = creationCodeInsertResult.rows[0].code_hash;
+    }
+  }
```

Comment:
If we would pass the `sourcifyDatabaseService` instead of the `poolClient`, we could call `sourcifyDatabaseService.database.insertCode()` directly. (would also need to move the withTransaction call)